src/diagnostics/core_diagnostics.jl

```diff
@@ -733,3 +733,341 @@ add_diagnostic_variable!(
     comments = "Elevation of the horizontal coordinates",
     compute! = compute_orog!,
 )
+
+###
+# cloud liquid water specific humidity (1d)
+###
+compute_clwvi!(out, state, cache, time) =
+    compute_clwvi!(out, state, cache, time, cache.atmos.moisture_model)
+compute_clwvi!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("clwvi", model)
+
+function compute_clwvi!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        Operators.column_integral_definite!(
+            out,
+            state.c.ρ .* cache.precomputed.cloud_diagnostics_tuple.q_liq,
+        )
+        return out
+    else
+        Operators.column_integral_definite!(
+            out,
+            state.c.ρ .* cache.precomputed.cloud_diagnostics_tuple.q_liq,
+        )
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "clwvi",
+    long_name = "Condensed Water Path",
+    standard_name = "atmosphere_mass_content_of_cloud_condensed_water",
+    units = "kg m-2",
+    comments = """
+    Mass of condensed (liquid + ice) water in the column divided by the area of the column 
+    (not just the area of the cloudy portion of the column). Includes precipitating hydrometeors
+    ONLY if the precipitating hydrometeors affect the calculation of radiative transfer in model.
+    """,
+    compute! = compute_clwvi!,
+)
+
+###
+# Dry Static Energy (1d)
+###
+compute_dsevi!(out, state, cache, time) =
+    compute_dsevi!(out, state, cache, time, cache.atmos.moisture_model)
+compute_dsevi!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("dsevi", model)
+
+function compute_dsevi!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+
+        c = CAP.cp_d(cache.params)
+        z = Fields.coordinate_field(state.c.ρ).z
+        g = CAP.grav(cache.params)
+        Operators.column_integral_definite!(
+            out,
+            Base.broadcasted(
+                *,
+                state.c.ρ,
+                Base.broadcasted(
+                    +,
+                    Base.broadcasted(
+                        *,
+                        c,
+                        Base.broadcasted(
+                            TD.air_temperature,
+                            thermo_params,
+                            cache.precomputed.ᶜts,
+                        ),
+                    ),
+                    Base.broadcasted(*, g, z),
```

Comment:
This code is not really readable. It would be great if, before merging, this code could be cleaned up in such a way that users not deeply familiar with the julia language could understand what is being computed here. If this requires changes in ClimaCore, let's apply those changes so that we can provide more user friendly interfaces.

Reply:
You can rewrite this to look like standard broadcasting using a macro---either Charlie's `@lazy_broadcasted` from LazyBroadcast.jl or my `@lazy_dot` from FusibleBroadcasts.jl

Reply:
I'm also going to turn accumulations and reductions into ClimaCore operators in a future release, which should make this all much simpler. Passing around unmaterialized broadcast objects is just an interim solution.

Reply:
Agreed - will look into it. Note to also change `c` to `cp`

Reply:
> I'm also going to turn accumulations and reductions into ClimaCore operators in a future release, which should make this all much simpler. Passing around unmaterialized broadcast objects is just an interim solution.

Thank you! Please, do not write public interfaces that require advanced knowledge of Julia to be used (so, keep all the Broadcasted machinery internal).